cpp/src_prims/stats/minmax.cuh

```diff
@@ -39,26 +40,36 @@ struct encode_traits<double> {
 
 HDI int encode(float val)
 {
-  int i = *(int*)&val;
+  int i{};
+  static_assert(sizeof(i) == sizeof(val));
```

Comment:
Maybe have something similar to `std::bit_cast`?

Reply:
Sure, maybe cuML needs something. But I am not aiming to solve *all* of cuML's problems in this single PR.

Reply:
Added `detail::bit_cast` in this file. Can be promoted if needed.